packages/quantic/cypress/integration/recent-queries-list/recent-queries-list.cypress.ts

```diff
@@ -63,38 +73,72 @@ describe('quantic-recent-queries-list', () => {
           Expect.displayQueries(true);
           Expect.numberOfQueries(index + 1);
           Expect.lastQueryContains(query);
-
+          Expect.displayQuery(query, true);
+          Expect.urlHashContains(query);
           clearInput();
         });
       });
 
       scope('when searching queries greater than #maxLength', () => {
-        createRandomQueriesList(15).forEach((query) => {
+        const listQueries = createRandomQueriesList(15);
+        listQueries.forEach((query) => {
           setQuery(query);
           performSearch().wait(InterceptAliases.Search);
           clearInput();
         });
 
         Expect.numberOfQueries(defaultMaxLength);
+        Expect.displayQuery(listQueries[0], false);
+      });
+
+      scope('when selecting a query in the list', () => {
+        const query1 = 'queryOne';
+        const query2 = 'queryTwo';
+
+        setQuery(query1);
+        performSearch().wait(InterceptAliases.Search);
+        clearInput();
+        setQuery(query2);
+        performSearch().wait(InterceptAliases.Search);
+        clearInput();
+
+        Expect.lastQueryContains(query2);
+        Expect.urlHashContains(query2);
+        RecentQueriesListSelectors.query(query1).click();
+        Expect.urlHashContains(query1);
       });
     });
   });
 
-  describe('with custom options', () => {
+  describe('with custom #maxLength', () => {
     const customMaxLength = 3;
 
     it('should work as expected', () => {
       visitRecentQueries({
         maxLength: customMaxLength,
       });
 
-      createRandomQueriesList(5).forEach((query) => {
+      const listQueries = createRandomQueriesList(5);
```

Comment:
We're pushing 5 queries and the list is showing only 3. The 2 first queries are not shown, but the test is only validating visibility for the first one.

I would suggest to push 4 queries instead of 5. The test would validate the same thing, but would be a little more strict. If you still prefer to push 5 queries, I would suggest to validate that the second one does not appear in the list at the end.

Reply:
I change it to 4 queries